semgrep-core/src/parsing/Parse_rule.ml

```diff
@@ -879,13 +779,47 @@ and parse_extra (env : env) (key : key) (value : G.expr) : extra =
       MetavarComparison { metavariable; comparison; strip; base }
   | _ -> error_at_key env key ("wrong parse_extra field: " ^ fst key)
 
-and parse_formula_old_from_dict (env : env) (rule_dict : dict) : R.formula =
-  let formula = parse_pair_old env (find_formula_old env rule_dict) in
+(*****************************************************************************)
+(* Parser for new  formula *)
+(*****************************************************************************)
+
+let find_formula env (rule_dict : dict) : key * G.expr =
+  let find key_str = Hashtbl.find_opt rule_dict.h key_str in
+  match
+    find_some_opt find
+      [ "pattern"; "and"; "or"; "regex"; "taint"; "not"; "inside" ]
+  with
+  | None ->
+      error env rule_dict.first_tok
```

Comment:
This error message refers to the "old" syntax. Also, the PR adding the `taint` operator has not yet been merged.